src/shared/utils/encryption/encryption.ts

```diff
@@ -61,3 +71,43 @@ export const getAppletEncryptionInfo = ({
 
   return key;
 };
+
+export const getAESKey = (
+  appletPrivateKey: number[],
+  userPublicKey: number[],
+  appletPrime: number[],
+  base: number[],
+) => {
+  const key = crypto.createDiffieHellman(Buffer.from(appletPrime), Buffer.from(base));
+  key.setPrivateKey(Buffer.from(appletPrivateKey));
+
+  const secretKey = key.computeSecret(Buffer.from(userPublicKey));
+
+  return crypto.createHash('sha256').update(secretKey).digest();
+};
+
+export const decryptData = ({ text, key }: { text: string; key: number[] }) => {
+  const textParts = text.split(':');
+  const iv = Buffer.from(textParts.shift()!, 'hex');
```

Comment:
please avoid using such short variable names

Reply:
I coped/pasted ecrypt/decrypt function from previous application as is, but it makes sense to update namings